pygbm/gradient_boosting.py

```diff
@@ -40,7 +40,7 @@ def fit(self, X, y):
         # TODO: add support for mixed-typed (numerical + categorical) data
         # TODO: add support for missing data
         # TODO: add support for pre-binned data (pass-through)?
-        X, y = check_X_y(X, y, dtype=[np.float32, np.float64])
```

Comment:
Why was this float64 BTW?

All the gradients / hessians structures are float32

Reply:
`dtype=[type_1, type_2]` means that X can either be of `type_1` or `type_2` and if it is not, it will be converted to `type_1`.